utils/slice_test.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2023 OWASP ModSecurity Core Rule Set Project
+// SPDX-License-Identifier: Apache-2.0
+
+package utils
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type sliceTestSuite struct {
+	suite.Suite
+}
+
+func TestSliceTestSuite(t *testing.T) {
+	suite.Run(t, new(timeTestSuite))
+}
+
+func (s *timeTestSuite) TestMatchSlice() {
+	re := regexp.MustCompile("^cookie$")
+
+	s.True(MatchSlice(re, []string{"cookie", "php"}))
+	s.False(MatchSlice(re, []string{"cooke", "php"}))
+	s.False(MatchSlice(re, []string{"cookies", "php"}))
```

Comment:
Add tests for
- the empty slice
- a slice with empty strings
- a matching tag that is not in the first position